### PR TITLE
TFKUBE-232: product version doc

### PR DESCRIPTION
**Pull request by yzha645**
Added documentation for variables that are related to product version and resource configuration.

### CODE REVIEW

File: config.tfvars [hunk 1]
```diff
@@ -4,23 +4,37 @@
 
 # Please define the values to configure the infrastructure before install
```
Comment (nghazali) [inline config.tfvars:5]: We may remove this line as we have similar content in line #2

File: config.tfvars [hunk 2]
```diff
@@ -4,23 +4,37 @@
...
+# To avoid storing license in a plain text file, we recommend storing it in an environment variable prefixed with `TF_VAR_` (i.e. `TF_VAR_bamboo_license`) and keep the below line commented out
+# If storing license as plain-text is not a concern for this environment, feel free to uncomment the following line and supply the license here
+#
+#bamboo_license = "<license key>"
 
-# Instance types that is preferred for node group.
-instance_types = ["m5.4xlarge"]
+# Bamboo system admin credentials
+# WARNING: In case you are restoring an existing dataset (see the `dataset_url` property below), you will need to use
+# credentials existing in the dataset. In this case, the `bamboo_admin_*` properties below are ignored.
```
Comment (nghazali) [inline config.tfvars:26]: We had some change here as part of TFKUBE-234 which is not here. Would you please pull the latest changes in main?

File: config.tfvars [hunk 3]
```diff
@@ -4,23 +4,37 @@
...
 
-# Custom tags for all resources to be created. Please add all tags you need to propagate among the resources.
-resource_tags = {
-  Terraform = "true"
-}
+# Bamboo license
+# To avoid storing license in a plain text file, we recommend storing it in an environment variable prefixed with `TF_VAR_` (i.e. `TF_VAR_bamboo_license`) and keep the below line commented out
+# If storing license as plain-text is not a concern for this environment, feel free to uncomment the following line and supply the license here
+#
+#bamboo_license = "<license key>"
```
Comment (nghazali) [inline config.tfvars:22]: Can we change the license key in upper case to match with others : `<LICENSE KEY>`

File: config.tfvars [hunk 4]
```diff
@@ -4,23 +4,37 @@
...
 
-# Instance types that is preferred for node group.
-instance_types = ["m5.4xlarge"]
+# Bamboo system admin credentials
+# WARNING: In case you are restoring an existing dataset (see the `dataset_url` property below), you will need to use
+# credentials existing in the dataset. In this case, the `bamboo_admin_*` properties below are ignored.
+bamboo_admin_username = "<USERNAME>"
+# To avoid storing system admin password in a plain text file, we recommend storing it in an environment variable prefixed with `TF_VAR_` (i.e. `TF_VAR_bamboo_admin_password`) and keep the below line commented out
+# If storing password as plain-text is not a concern for this environment, feel free to uncomment the following line and supply system admin password here
+#bamboo_admin_password      = "<password>"
```
Comment (nghazali) [inline config.tfvars:30]: change to uppercase => `<PASSWORD>`

File: config.tfvars [hunk 5]
```diff
@@ -29,6 +43,28 @@ desired_capacity = 2
...
 
+# (Optional) URL for dataset to import
+# The provided default is the dataset used in the DCAPT framework.
+# See https://developer.atlassian.com/platform/marketplace/dc-apps-performance-toolkit-user-guide-bamboo/#2--preloading-your-bamboo-deployment-with-an-enterprise-scale-dataset
+# for details
+#
+#dataset_url = "https://centaurus-datasets.s3.amazonaws.com/bamboo/dcapt-bamboo.zip"
+
+
+################################################################################
```
Comment (nghazali) [inline config.tfvars:54]: I think we can remove `Optional Configurations` title. All following items could go under Advanced.

**Review by nghazali — CHANGES_REQUESTED** [review]

File: docs/docs/userguide/CONFIGURATION.md [hunk 6]
```diff
@@ -219,6 +219,42 @@ db_iops = 1000
...
+
+`bamboo_helm_chart_version` sets the [Helm chart](https://github.com/atlassian/data-center-helm-charts){.external} version of Bamboo Agent instance.
+
+```terraform
+bamboo_agent_helm_chart_version = "1.0.0"
+```
+
+### Bamboo instance resource configuration
+
+The following variables set number of cpu, amount of memory, maximum heap size and minimum heap size of Bamboo instance. (Used default values as example.)
```
Comment (nghazali) [inline docs/docs/userguide/CONFIGURATION.md:240]:
```suggestion
The following variables set number of CPU, amount of memory, maximum heap size and minimum heap size of Bamboo instance. (Used default values as example.)
```

File: docs/docs/userguide/CONFIGURATION.md [hunk 7]
```diff
@@ -219,6 +219,42 @@ db_iops = 1000
...
+```terraform
+bamboo_cpu = "1"
+bamboo_mem = "1Gi"
+bamboo_min_heap = "256m"
+bamboo_max_heap = "512m"
+```
+
+### Bamboo Agent instance resource configuration
+
+The following variables set number of cpu and amount of memory of Bamboo Agent instances. (Used default values as example.)
```
Comment (nghazali) [inline docs/docs/userguide/CONFIGURATION.md:251]:
```suggestion
The following variables set number of CPU and amount of memory of Bamboo Agent instances. (Used default values as example.)
```

File: config.tfvars (hunk 3 above)
Comment (jjeongatl) [reply config.tfvars:22]: Updating this on https://github.com/atlassian-labs/data-center-terraform/pull/113

File: config.tfvars (hunk 4 above)
Comment (jjeongatl) [reply config.tfvars:30]: Updating this on https://github.com/atlassian-labs/data-center-terraform/pull/113

File: README.md [hunk 8]
```diff
@@ -4,7 +4,7 @@
 **NOTE: This project is under heavy development, and it is not officially supported yet.**
 
 Atlassian DC APP program provides App vendors in our ecosystem with ready-to-use environment setup tools. 
-This project provides a tool to provision infrastructure for Atlassian DC helm chart products.
+This project provides a tool to provision infrastructure for Atlassian DC Helm chart products.
```
Comment (nanux) [inline README.md:7]:
```suggestion
This project provides a tool to provision infrastructure for Atlassian DC products deployed with [Helm charts](https://atlassian.github.io/data-center-helm-charts/).
```

File: config.tfvars [hunk 9]
```diff
@@ -2,7 +2,9 @@
 # Please configure this file carefully before installing the infrastructure.
 # See https://github.com/atlassian-labs/data-center-terraform/blob/main/README.md for more information.
```
Comment (nanux) [inline config.tfvars:3]:
```suggestion
# See https://atlassian-labs.github.io/data-center-terraform/userguide/CONFIGURATION/ for more information.
```

**Review by nanux — CHANGES_REQUESTED** [review]

**Review by nghazali — APPROVED** [review]

**Review by nghazali — APPROVED** [review]

**Review by nanux — APPROVED** [review]

**Pull request merged**